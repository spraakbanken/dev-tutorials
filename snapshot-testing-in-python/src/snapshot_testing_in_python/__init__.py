from dataclasses import dataclass
from typing import Any


def hello() -> str:
    return "Hello from snapshot-testing-in-python!"


@dataclass
class User:
    name: str
    timestamp: int

    def to_dict(self) -> dict[str, Any]:
        return self.__dict__
