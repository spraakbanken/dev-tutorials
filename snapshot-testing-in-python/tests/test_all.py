import random

import pytest
from snapshot_testing_in_python import User, hello
from syrupy.extensions.json import JSONSnapshotExtension
from syrupy.matchers import path_type


@pytest.fixture
def snapshot_json(snapshot):
    return snapshot.use_extension(JSONSnapshotExtension)


def test_hello(snapshot) -> None:
    assert hello() == snapshot


def random_user() -> User:
    return User(name="Mormor Karl", timestamp=random.randint(100000, 200000))


def test_user(snapshot_json) -> None:
    user = random_user()

    assert user.to_dict() == snapshot_json(matcher=path_type({"timestamp": (int,)}))
