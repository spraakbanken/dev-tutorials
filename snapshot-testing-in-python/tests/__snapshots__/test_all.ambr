# serializer version: 1
# name: test_hello
  'Hello from snapshot-testing-in-python!'
# ---
